### ISSUE TITLE
Core validation incorrect for vkCmdClearAttachments

### ISSUE THREAD

**Issue by gary-sweet** [KhronosGroup/Vulkan-LoaderAndValidationLayers#919]
When the core validation layer is validating that an attachment is in the reference list of active subpasses it doesn't take into account the extra indirection required by the subpass description array.

The following code fragment contains the issue:
```
for (uint32_t attachment_idx = 0; attachment_idx < attachmentCount; attachment_idx++) {
    const VkClearAttachment *attachment = &pAttachments[attachment_idx];
    ...
    for (uint32_t i = 0; i < pSD->colorAttachmentCount; i++) {
       if (attachment->colorAttachment == pSD->pColorAttachments[i].attachment) {
          found = true;
```

This is testing that the value of the attachment->colorAttachment field of the VkClearAttachment exists in the subpass description list for the subpass.

The spec states that the colorAttachment field `is an index to the pColorAttachments array in the VkSubpassDescription structure of the current subpass`.  The correct tests here would be that attachment->colorAttachment was smaller than the size of the pSD->pColorAttachments array. You might also then test that the attachment referenced by the pSD->pColorAttachment at that index exists in the render pass attachment list.

My tests which found this problem make a render pass with many attachments (up to 20). Each subpass uses a small subset of those, and crucially, not in contiguous order. So the first subpass might use attachments 1, 4, 6 & 9 for example. The validation code as it is today complains about this usage, which I believe to be valid.

**Comment by chrisforbes** [comment]
I agree with your analysis.

**Comment by chrisforbes** [comment]
We should also be downgrading these errors to warnings-- the spec defines the behavior for out of range attachment indices, and use of VK_ATTACHMENT_UNUSED. I will fix this at the same time.

**Comment by chrisforbes** [comment]
Fixed in #923

**Comment by gary-sweet** [comment]
Great, thanks. I've pulled the fix and it fixes my issue locally.
